appeals/web/src/server/appeals/appeal-details/appellant-case/appellant-case.mapper.js

```diff
@@ -19,41 +19,43 @@ export function mapResponseToSummaryListBuilderParameters(appellantCaseData) {
 /**
  *
  * @param {import('./appellant-case.service.js').AppellantCaseReviewOutcome} reviewOutcome
- * @param {import('@pins/appeals.api').Schema.AppellantCaseInvalidReason[]} invalidReasonOptions
+ * @param {import('appeals/api/src/database/schema.js').AppellantCaseInvalidReason[]|import('appeals/api/src/database/schema.js').AppellantCaseIncompleteReason[]} invalidOrIncompleteReasonOptions
+ * @param {string} [updatedDueDate]
  * @returns {SummaryListBuilderParameters}
  */
 export function mapReviewOutcomeToSummaryListBuilderParameters(
 	reviewOutcome,
-	invalidReasonOptions
+	invalidOrIncompleteReasonOptions,
+	updatedDueDate
 ) {
+	if (reviewOutcome.validationOutcome === 'invalid' && !reviewOutcome.invalidReasons) {
+		throw new Error('validationOutcome is "invalid" but invalidReasons was not found');
+	}
+	if (reviewOutcome.validationOutcome === 'incomplete' && !reviewOutcome.incompleteReasons) {
+		throw new Error('validationOutcome is "incomplete" but incompleteReasons was not found');
+	}
+
+	/**
+	 * @type {string[]}
+	 */
 	const reasonsContent = [];
 
-	switch (reviewOutcome.validationOutcome) {
-		case 'invalid':
-			if (!reviewOutcome.invalidReasons) {
-				throw new Error('validationOutcome is "invalid" but invalidReasons was not found');
+	(reviewOutcome.validationOutcome === 'invalid'
+		? reviewOutcome.invalidReasons
+		: reviewOutcome.incompleteReasons
+	)?.forEach((reason) => {
+		for (const possibleReason of invalidOrIncompleteReasonOptions) {
```

Comment:
I think this logic would be clearer with a .filter().map call, rather than pushing to an array in a forEach.

Reply:
I've updated this to use map and filter, personally I find it less clear this way, but I suppose it's a matter of opinion

Reply:
I agree with you, I perhaps misunderstood the logic. I thought a filter could be used to remove reasons without a reason option, but it looks like it needs to error instead. I do think the map makes the intention clearer, that each reason will map to an option.

This is a possible option to make it more readable, interested in what you think:

```
const otherNotValidReasons = reviewOutcome.otherNotValidReasons || '';
const reasons = reviewOutcome.validationOutcome === appellantCaseReviewOutcomes.invalid
	? reviewOutcome.invalidReasons
	: reviewOutcome.incompleteReasons;

const reasonsContent = reasons
	?.map(reason => invalidOrIncompleteReasonOptions.find(option => option.id === reason))
	.map(option => {
		if (!option) {
			throw new Error('invalid or incomplete reason ID was not recognised');
		}
		const name = option.name;
		return name.toLowerCase() === 'other' ? `${name}: ${otherNotValidReasons}` : name;
	}) || [''];
```

No need to change though, it's not a big deal for now.